app/src/main/java/mn/devfest/sessions/details/SessionDetailsFragment.java

```diff
@@ -98,12 +99,29 @@ public void onViewCreated(View view, @Nullable Bundle savedInstanceState) {
         super.onViewCreated(view, savedInstanceState);
         ButterKnife.bind(this, view);
 
-        //Set the session member variable
+        //Set the session ID member variable
         Bundle args = getArguments();
         if (args != null && args.containsKey(ARG_SESSION_ID)) {
             //Get session from data layer
-            String sessionId = args.getString(ARG_SESSION_ID);
-            mSession = mDataSource.getSessionById(sessionId);
+             mSessionId = args.getString(ARG_SESSION_ID);
+            if (mSessionId != null) {
+                mFirebaseDatabaseReference = FirebaseDatabase.getInstance().getReference();
+                mFirebaseDatabaseReference.child(DEVFEST_2017_KEY).child(SESSIONS_CHILD_KEY).child(mSessionId)
```

Comment:
Same deal about putting Firebase logic in a dedicated class for each part of the Firebase data store to be monitored.